keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1319 @@
+3.  Provide a CCI test driver which can demonstrate CCI Driver implementation requirementes and several resource management use-cases for cpu and memory.
+4.  Provide documntation of the CCI Manager and the provided test driver plus illustration of the currenlty covered use-cases.
+5.	Support seamless k8s start.
+6.	CCI Driver results (cpusets and memory affinity) are passed back to kubelet and allocated.
+7.	Handle resource management state for cpu and memory through CCI. 
+8.	Support proper fail mechanisms of Pods if CCI driver is not available - Error message + retry 
+9.	Continue cluster operation if CCI driver is not available
+10.  Podspecs:  Will be able to support current pod specs.  While there may be additional 
+extensibility in the future, the current pod specs will still work.
+
+
+#### Beta & Post-Beta Goals
+
+1.	Support standard pod deployments not requiring CCI Drivers through CPU & Memory Manager: all QoS types
+2.  Support hint providers for topology manager
+3.  Interoperability with Device Plugins, DRA, et cetera. 
+4.	Identify and provide a minimal in-cluster operational core (from existing CPU Manager, Memory Manager, Topology Manager) either through in-tree libraries or through further KEPs. 
+5.  Replace CCI policy with a component integrated in None and Static CPU Manager policies.
+6.  Gather feedback on the feature through surveys.
+7.	E2E testing including other components.
+8.  Adding support for e2e tests in testgrid.
+9.  Health identification mechanisms and automatic health-repair procedures for CCI Drivers.  
+8.  Optimize on-node resource usage (approaches to avoid waste of resources).
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Break any existing use cases for topology manager, memory manager, cpu manager, or 
+device manager.
+* Creating any more latency than there is today for scheduling:  We should be 
+careful not to add any latency into scheduling over what exists today for default behavior.
+* CPU & Memory Management on scheduler side.
+* Optimal binpacking of resources (we will still consider if we can optimize resource utilization in Beta).
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+
+#### Custom workloads, such as Telco/HPC/AI/ML
```

Comment:
Thanks for providing the use cases that we would like to enable with this feature in detail. This nicely covers the "why" of the feature. I think in this user stories section we should cover "who" as well for this feature and explicitly state the responsibilities of the key actors: cluster operator, hardware vendor and end user. Please refer to user stories section in [device plugin KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3573-device-plugin/README.md#user-stories) and [DRA KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3063-dynamic-resource-allocation#user-stories) as examples.

Reply:
Added, I think.  Please make sure this is sufficient, as this is user-level.

Reply:
Thanks for the additions but we need to clearly highlight responsibilities of cluster operator, hardware vendor and end user in terms of what they do to enable/utilize this feature. I don't see that in the text right now. Something like:

1. Cluster operator/sys admin has to enable the right feature gates and ensure that kubelet is correctly configured, CCIDriver is deployed, perhaps ResourceClass creation?
2. Hardware Vendor writes the CCIDriver...
3. User specifies resourceClaim in the pod spec...

If we are following similar mechanism as DRA, are supporting the creation of ResourceClaimTemplates? Some examples for each of these specifications would be useful and help with better understanding of the feature.

Reply:
done